semcore/ellipsis/src/Ellipsis.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useRef, useState } from 'react';
+import createComponent, { Component, sstyled } from '@semcore/core';
+import Tooltip from '@semcore/tooltip';
+import { Flex } from '@semcore/flex-box';
+import { useResizeObserver } from './useResizeObserver';
+
+import style from './style/ellipsis.shadow.css';
+import reactToText from '@semcore/utils/lib/reactToText';
+import getOriginChildren from '@semcore/utils/lib/getOriginChildren';
+
+type AsProps = {
+  maxline?: number;
+  trim?: string;
+  tooltip?: string;
+  styles?: React.CSSProperties;
+  resizeObserver: () => void;
+};
+
+type AsPropsMiddle = {
+  text?: string;
+  tooltip?: string;
+  styles?: React.CSSProperties;
+  resizeObserver: () => void;
+};
+
+const gotSize = () => {
```

Comment:
What that function is doing and why size is already got?

Reply:
it counts the size of one character